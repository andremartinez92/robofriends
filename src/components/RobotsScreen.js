// @flow
import React from 'react';

import Header from '../components/Header';
import SearchBox from '../components/SearchBox';
import RobotsList from '../components/RobotsList';

import { type RobotData } from '../types';

import './RobotsScreen.css';

type Props = {
    isPending: boolean,
    onSearchChange: (searchField: string) => void,
    robots: RobotData[],
}

function RobotsScreen({
    isPending,
    onSearchChange,
    robots
}: Props) {
    return (
        <div className="tc">
            <Header />
            <SearchBox onSearchChange={onSearchChange} />
            <RobotsList isPending={isPending} robots={robots} />
        </div>
    );
}

export default RobotsScreen;