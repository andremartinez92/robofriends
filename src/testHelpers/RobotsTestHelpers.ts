// @flow
/* eslint-disable no-useless-concat */
import { RobotData } from '../types';

export const TITLE_TEXT = 'Robofriends';
export const LOADING_TEXT = 'Loading...';
export const NO_DATA_TEXT = 'Could not retrieve content' + 'Please check your internet connection.';

export const ROBOTS: RobotData[] = [
    {
        email: 'something@email.com',
        id: 1,
        name: 'Robo 1',
        username: 'robo1'
    },
    {
        email: 'random@email.com',
        id: 2,
        name: 'Robo 2',
        username: 'robo2'
    },
]

export const ROBO1_TEXT = 'Robo 1' + 'something@email.com';
export const ROBO2_TEXT = 'Robo 2' + 'random@email.com';
export const ROBOTS_TEXT = ROBO1_TEXT + ROBO2_TEXT;