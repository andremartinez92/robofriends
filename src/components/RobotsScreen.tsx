import React from 'react';

import Header from './Header';
import SearchBox from './SearchBox';
import RobotsList from './RobotsList';

import { RobotData } from '../types';

import './RobotsScreen.css';

interface Props {
    isPending: boolean,
    onSearchChange: (searchField: string) => void,
    robots: RobotData[],
}

function RobotsScreen({
    isPending,
    onSearchChange,
    robots
}: Props) {
    return (
        <div className="tc">
            <Header />
            <SearchBox onSearchChange={onSearchChange} />
            <RobotsList isPending={isPending} robots={robots} />
        </div>
    );
}

export default RobotsScreen;