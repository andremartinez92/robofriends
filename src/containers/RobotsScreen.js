// @flow

import React from 'react';

import './RobotsScreen.css';
import { type RobotData } from '../types';
import CardList from '../components/CardList';
import Header from '../components/Header';
import SearchBox from '../components/SearchBox';
import ScrollableSection from '../components/ScrollableSection';
import ErrorBoundary from '../components/ErrorBoundary';
import filterRobotsBySearch from '../logic/filterRobotsBySearch';

function RobotsScreen() {
  const [robots, setRobots] = React.useState<RobotData[]>([]);
  const [searchField, setSearchField] = React.useState('');

  React.useEffect(() => {
    async function fetchData() {
      const response = await fetch(
        'https://jsonplaceholder.typicode.com/users'
      );
      const robots = await response.json();
      setRobots(robots);
    }

    fetchData();
  }, []);

  const filteredRobots: RobotData[] = filterRobotsBySearch(robots, searchField);

  return (
    <div className="tc">
      <Header />
      <SearchBox onSearchChange={event => setSearchField(event.target.value)} />
      <ScrollableSection>
        <ErrorBoundary>
          <CardList robots={filteredRobots} />
        </ErrorBoundary>
      </ScrollableSection>
    </div>
  );
}

export default RobotsScreen;
