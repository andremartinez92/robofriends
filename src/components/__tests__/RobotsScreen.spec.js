import React from 'react';
import { shallow } from 'enzyme';
import { ROBOTS } from '../../testHelpers/RobotsTestHelpers';

import RobotsScreen from '../RobotsScreen';

describe('RobotsScreen', () => {
  it('renders correctly when loading', () => {
    const wrapper = shallow(<RobotsScreen isPending robots={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when there are no robots', () => {
    const wrapper = shallow(<RobotsScreen isPending={false} robots={[]} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when there are robots', () => {
    const wrapper = shallow(<RobotsScreen isPending={false} robots={ROBOTS} />);
    expect(wrapper).toMatchSnapshot();
  });
});
