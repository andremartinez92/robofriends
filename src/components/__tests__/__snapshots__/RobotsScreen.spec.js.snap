// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RobotsScreen renders correctly when loading 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <RobotsList
    isPending={true}
    robots={Array []}
  />
</div>
`;

exports[`RobotsScreen renders correctly when there are no robots 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <RobotsList
    isPending={false}
    robots={Array []}
  />
</div>
`;

exports[`RobotsScreen renders correctly when there are robots 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <RobotsList
    isPending={false}
    robots={
      Array [
        Object {
          "email": "something@email.com",
          "id": 1,
          "name": "Robo 1",
          "username": "robo1",
        },
        Object {
          "email": "random@email.com",
          "id": 2,
          "name": "Robo 2",
          "username": "robo2",
        },
      ]
    }
  />
</div>
`;
